qcodes_contrib_drivers/drivers/Bilt/ITest.py

```diff
@@ -0,0 +1,467 @@
+        
+        if isinstance(outf,(int,float)):
+            mode = str(outf)
+        else:
+            if outf=='exp':
+                mode = '0'
+            elif outf=='ramp':
+                mode = '1'
+        
+        self.write(chan_id + 'trig:input ' + mode)
+
+
+    def _get_output_function(self, chan:int) -> str:
+        """
+        Get output volage update  function
+        
+        Args:
+            chan: The 1-indexed channel number
+        
+        Returns:
+            mode
+        """
+        chan_id = self.chan_to_id(chan)
+        mode = int(self.ask(chan_id + 'TRIG:INPUT?'))
+        if mode == 0:
+            return 'exp'
+        elif mode == 1:
+            return 'ramp'
+        else:
+            return 'ERROR'
+
+
+    def _set_chan_range(self, chan:int,
+                              volt: float) -> None:
+        """
+        Set output voltage range
+        
+        Args:
+            chan : The 1-indexed channel number
+            volt : Voltage range (1.2 or 12)
+        """
+        chan_id = self.chan_to_id(chan)
+        self.write(chan_id + 'VOLT:RANGE ' + str(volt))
+
+
+    def _get_chan_range(self, chan:int) -> str:
+        """
+        Get output voltage range
+        
+        Args:
+            chan: The 1-indexed channel number
+            
+        Returns:
+            volt: Output voltage range
+        """
+        chan_id = self.chan_to_id(chan)
+        
+        return self.ask(chan_id + 'VOLT:RANGE?')[:-2]
+
+
+    def _set_chan_pos_sat(self, chan:int,
+                                pos_sat: Union[int, str]) -> None:
+        chan_id = self.chan_to_id(chan)
+        if isinstance(pos_sat,(int,float)):
+            self.write(chan_id + 'VOLT:SAT:POS {:.8f}'.format(pos_sat))
+        elif isinstance(pos_sat,str):
+            self.write(chan_id + 'VOLT:SAT:POS MAX')
+
+
+    def _set_chan_neg_sat(self, chan:int,
+                                neg_sat: Union[int, str]) -> None:
+        chan_id = self.chan_to_id(chan)
+        if isinstance(neg_sat,(int,float)):
+            self.write(chan_id + 'VOLT:SAT:NEG {:.8f}'.format(neg_sat))
+        elif isinstance(neg_sat,str):
+            self.write(chan_id + 'VOLT:SAT:NEG MIN')
+
+
+    def _get_chan_pos_sat(self, chan:int) -> str:
+        chan_id = self.chan_to_id(chan)
+        return self.ask(chan_id + 'VOLT:SAT:POS ?')
+
+
+    def _get_chan_neg_sat(self, chan:int) -> str:
+        chan_id = self.chan_to_id(chan)
+        return self.ask(chan_id + 'VOLT:SAT:NEG ?')
+
+
+    def _get_chan_state(self, chan:int) -> bool:
+        """
+        Get channel power state
+        
+        Args:
+            chan: The 1-indexed channel number
+            
+        Returns:
+            state: Power state 
+        """
+        chan_id = self.chan_to_id(chan)
+        state = self.ask(chan_id + 'OUTP ?')
+        
+        if state == '1':
+            return True
+        elif state == '0':
+            return False
```

Comment:
What do you expect if state is not '1' or '0'? can they be other than these two options?

Reply:
I do not expect `state` to be anything different.
However I added an `else` statement with a error in case of.